semantic-ast/src/AST/Unmarshal.hs

```diff
@@ -60,10 +59,6 @@ import           TreeSitter.Node as TS
 import           TreeSitter.Parser as TS
 import           TreeSitter.Tree as TS
 
-asks :: Has (Reader r) sig m => (r -> r') -> m r'
-asks f = send (Ask (pure . f))
-{-# INLINE asks #-}
-
```

Comment:
`asks` [is now inlined in `fused-effects` 1.1](http://hackage.haskell.org/package/fused-effects-1.1.0.0/docs/src/Control.Effect.Reader.html#asks), so we don’t have to provide our own definition.

Note that it is _extremely_ likely that this was a red herring in the profiles in the first place; `-fprof-auto` inserts virtual `SCC` annotations around every call to a non-`INLINE`d function, which (among other things) artificially prevents inlining from taking place, and slows down the resulting program. Thus a program which would have cheaply computed a bunch of `ask`s by simply _returning a parameter_ can end up looking extremely expensive in the profile simply because we’ve asked `ghc` to do something a bit silly.

Incidentally, this is contributing significantly to my general distaste for `ghc`’s extant profiling mechanisms.